submissions/andrewklmn/a-tiny-js-world/index.js

```diff
@@ -0,0 +1,102 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/andrewklmn/a-tiny-JS-world
+   Web app: https://andrewklmn.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+const dog = {
+  species: 'dog',
+  name: 'Dyuka',
+  gender: 'male',
+  legs: 4,
+  hands: 0,
+  saying: 'woof!',
+  friends: [],
+};
+
+const cat = {
+  species: 'cat',
+  name: 'Barsik',
+  gender: 'male',
+  legs: 4,
+  hands: 0,
+  saying: 'meow!',
+  friends: [],
+};
+
+const woman = {
+  species: 'human',
+  name: 'Leeloo Dallas',
+  gender: 'female',
+  legs: 2,
+  hands: 2,
+  saying: 'People hi!',
+  friends: [],
+};
+
+const man = {
+  species: 'human',
+  name: 'Korben Dallas',
+  gender: 'male',
+  legs: 2,
+  hands: 2,
+  saying: 'Hello there!',
+  friends: [],
+};
+
+// define cat-woman
+const catWoman = { ...woman };
+catWoman.name = 'Cat-woman';
+catWoman.saying = cat.saying;
+
+// define friends for everyone except the cat...
+dog.friends = [man, woman, cat];
+woman.friends = [man, dog, cat];
+man.friends = [woman, dog, cat, catWoman];
+catWoman.friends = [cat];
+
+const inhabitants = [
+  dog,
+  cat,
+  woman,
+  catWoman,
+  man,
+];
+
+const getFormatedOutput = (obj) => Object.entries(obj).map((a) => {
```

Comment:
As for me, it is a hard function. But it is your choice. 
```
  const getFormatedOutput = ({species, name, gender, legs, hands, saying, friends}) =>  {
    if(typeof(friends) === 'object'){
        friends = friends.map(friend => friend.name).join(', ');
    } 

    return `${species}; ${name}; ${gender}; ${legs}; ${hands}; ${saying}; ${friends}`;
  }

```

Reply:
Such a cool solution!
I took it into my code.